File: Bai_84/main.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <cctype>

bool isStrongPassword(const std::string& password) {
    if (password.length() < 8) {
        return false;
    }
    bool hasDigit = false;
    bool hasUpper = false;
    bool hasLower = false;
    bool hasSpecial = false;
    for (char c : password) {
        if (isdigit(c)) {
            hasDigit = true;
        } else if (isupper(c)) {
            hasUpper = true;
        } else if (islower(c)) {
            hasLower = true;
        } else {
            hasSpecial = true;
        }
    }
    return hasDigit && hasUpper && hasLower && hasSpecial;
}

int main() {
    std::ifstream inputFile("BAI3.INP");
    std::string password;
    std::getline(inputFile, password);

    std::ofstream outputFile("BAI3.OUT");
    if (isStrongPassword(password)) {
        outputFile << "Tối ưu";
    } else {
        outputFile << "Không tối ưu";
    }

    return 0;
}

File: Bai_84/main.py
def is_strong_password(password):
    if len(password) < 8:
        return False
    has_digit = False
    has_upper = False
    has_lower = False
    has_special = False
    for char in password:
        if char.isdigit():
            has_digit = True
        elif char.isupper():
            has_upper = True
        elif char.islower():
            has_lower = True
        else:
            has_special = True
    return has_digit and has_upper and has_lower and has_special

# Đọc dữ liệu từ tệp
with open('BAI3.INP', 'r') as f:
    password = f.readline().strip()

# Kiểm tra mật khẩu
if is_strong_password(password):
    result = "Tối ưu"
else:
    result = "Không tối ưu"

# Ghi kết quả vào tệp
with open('BAI3.OUT', 'w', encoding='utf-8') as f:
    f.write(result)